File: sdk/src/resources/events.ts
import type { HttpClient } from "../http";
import type { ListEventsParams, ListEventsResult } from "../types";

export class EventsResource {
  constructor(private http: HttpClient) {}

  /**
   * Poll recent events (inbound messages, orders, new contacts).
   * Designed for webhook alternatives and Zapier-style integrations.
   *
   * Pass the returned `nextAfter` value back as `after` on the next
   * call to fetch only newer events — this is the cursor pattern.
   *
   * @example — simple poll
   * const { events, nextAfter } = await client.events.list({ type: "message.received" });
   *
   * @example — continuous polling loop
   * let after: string | null = null;
   * while (true) {
   *   const result = await client.events.list({ after: after ?? undefined });
   *   for (const event of result.events) processEvent(event);
   *   after = result.nextAfter;
   *   await new Promise(r => setTimeout(r, 60_000)); // poll every minute
   * }
   */
  async list(params: ListEventsParams = {}): Promise<ListEventsResult> {
    return this.http.get<ListEventsResult>("/events", { params: params as Record<string, unknown> });
  }
}

File: sdk-python/wappflow/resources/events.py
from __future__ import annotations

from typing import Optional

from .._client import HttpClient
from .._types import EventType, ListEventsResult


class EventsResource:
    def __init__(self, http: HttpClient) -> None:
        self._http = http

    def list(
        self,
        *,
        type: Optional[EventType] = None,
        after: Optional[str] = None,
        limit: int = 20,
    ) -> ListEventsResult:
        """Poll recent events.

        Pass the returned ``nextAfter`` value back as ``after`` on the next
        call to receive only newer events (cursor-based pagination).

        Requires scope ``contacts:read``.

        Event types:
            - ``"message.received"`` — inbound WhatsApp messages
            - ``"message.status"``   — delivery / read receipts
            - ``"order.placed"``     — WhatsApp store orders
            - ``"contact.created"``  — new CRM contacts

        Example::

            next_after = None
            while True:
                result = client.events.list(type="message.received", after=next_after)
                for event in result["events"]:
                    handle(event)
                next_after = result["nextAfter"]
                time.sleep(60)
        """
        return self._http.get(
            "/events",
            params={"type": type, "after": after, "limit": limit},
        )
